exporter/elasticsearchexporter/exporter.go

```diff
@@ -252,18 +172,18 @@ func (e *elasticsearchExporter) pushMetricsData(
 	ctx context.Context,
 	metrics pmetric.Metrics,
 ) error {
-	mappingMode := e.config.MappingMode()
+	mappingMode, err := e.getMappingMode(ctx)
+	if err != nil {
+		return err
```

Comment:
Shouldn't this be a permanent error?
```suggestion
		return consumererror.NewPermanent(err)
```

By the way, it looks like the Elasticsearch exporter does not use `consumererror.NewPermanent()` anywhere. Does this mean it never returns permanent errors to its callers? Perhaps something worth looking at?

Reply:
On a second thought, this is probably not a blocker for this PR, but rather a wider issue to look into, as the Elasticsearch exporter does not seem to make use of permanent errors anywhere.

Reply:
https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/38268

Reply:
Thanks @andrzej-stencel, agreed that we should look into that, and that it's a broader issue.